<?xml version="1.0"?>
<launch>
    <arg name="monodepth_flag" default="true"/>
    <arg name="monodepth_ckpt_file" default="/home/apoorv/workspaces/omnidepth_ws/src/omnidepth_ros2/models/metric_3d.onnx"/>
    <arg name="monodepth_gpu_index" default="0"/>
    <arg name="cam_channel_name" default="front"/>

    <arg name="camera_topic" default="/front_stereo_camera/right/image_raw"/>
    <arg name="compressed_image_topic" default="/image_raw/compressed"/>
    <arg name="camera_param" default="/front_stereo_camera/right/camera_info"/>
    <arg name="output_bbox_topic" default="/bboxes"/>
    <arg name="output_seg_pointcloud_topic" default="/point_cloud_front"/>
    <arg name="output_depth_image_topic" default="/depth_image_front"/>


    <node pkg="omnidepth_ros2" exec="omnidepth_inference_node" name="omnidepth_inference" output="screen">
        <param name="MONODEPTH_FLAG" value="$(var monodepth_flag)"/>
        <param name="MONODEPTH_CKPT_FILE" value="$(var monodepth_ckpt_file)"/>
        <param name="MONODEPTH_GPU_INDEX" value="$(var monodepth_gpu_index)"/>
        <param name="CAM_CHANNEL_NAME" value="$(var cam_channel_name)"/>

        <remap from="/image_raw" to="$(var camera_topic)"/>
        <remap from="/compressed_image" to="$(var compressed_image_topic)"/>
        <remap from="/camera_info" to="$(var camera_param)"/>
        <remap from="/bboxes" to="$(var output_bbox_topic)"/>
        <remap from="/point_cloud" to="$(var output_seg_pointcloud_topic)"/>
        <remap from="/depth_image" to="$(var output_depth_image_topic)"/>
    </node>

</launch>